### DISCUSSION TITLE
How to identify casks for M1 CPUs?

### DISCUSSION THREAD

**Discussion by FlavioB79** [Homebrew/discussions#5176] [Everyday usage]
### Output of `brew config`

```shell
bonny@Flavios-MacBook-Pro ~ % brew config
HOMEBREW_VERSION: 4.2.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: c6d959218f143cd17b1fc3e0f10f143cbd273528
Last commit: 8 days ago
Core tap JSON: 01 Mar 14:58 UTC
Core cask tap JSON: 01 Mar 14:58 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.3.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
bonny@Flavios-MacBook-Pro ~ % brew doctor
Your system is ready to brew.
```

### Description of issue

Hi all.
I'm about to migrate from my old MacBook Pro with Intel CPU to a more recent MacBook Pro with M1 CPU.
Now I've just run "brew install" with the list of software I had installed on my old MBP but some of them won't run. Maybe because of not being the M1-versions. Is there a way to identify these, so that I can install the correct versions?
Thanks,
F.

**Comment by kencu** [comment]
don’t forget to install Rosetta2 on your new M1 Mac, for maximal flexibility in running software.

It’s not installed by default.

**Reply by FlavioB79** [reply]
Oh, so I just have to install Rosetta2 and that's it?

**Reply by SMillerDev** [reply]
It really depends on the software. Some software only runs on Intel. Some software you can run on arm in Rosetta and some will just work when you run `brew install`

**Reply by FlavioB79** [reply]
OK Thanks!

**Reply by p-linnane** [reply]
Adding onto this: if there is a separate arm64 version of a cask, and we know about it, brew will install that automatically. You don't have to specify anything.

**Reply by FlavioB79** [reply]
Thanks for this additional info too.
